heroic-core/src/main/java/com/spotify/heroic/http/HttpServer.java

```diff
@@ -213,7 +213,7 @@ private HandlerCollection setupHandler() throws Exception {
 
         context.addServlet(jerseyServlet, "/*");
         context.addFilter(new FilterHolder(new ShutdownFilter(stopping, mapper)), "/*", null);
-        context.addFilter(new FilterHolder(new MandatoryClientIdFilter()), "/*", null);
+        context.addFilter(new FilterHolder(new MandatoryClientIdFilter(mapper)), "/*", null);
```

Comment:
Do we want to also require client ids for non `/query/*` endpoints?

Reply:
I believe we only want it for users querying metrics. Writes don't need a client id since the tags should be enough to tell you where it came from if that's the concern.

Reply:
I think that makes sense. I also have the sneaky feeling that writes haven't been tested at the system level and we really don't want those to fail...